docs/docs/module_guides/indexing/lpg_index_guide.md

```diff
@@ -0,0 +1,573 @@
+)
+
+# use
+retriever = index.as_retriever(
+    include_text=True,  # include source chunk with matching paths
+    similarity_top_k=2,  # top k for vector kg node retrieval
+)
+nodes = retriever.retrieve("Test")
+
+query_engine = index.as_query_engine(
+    include_text=True,  # include source chunk with matching paths
+    similarity_top_k=2,  # top k for vector kg node retrieval
+)
+response = query_engine.query("Test")
+
+# save and load
+index.storage_context.persist(persist_dir="./storage")
+
+from llama_index.core import StorageContext, load_index_from_storage
+
+index = load_index_from_storage(
+    StorageContext.from_defaults(persist_dir="./storage")
+)
+
+# loading from existing graph store (and optional vector store)
+# load from existing graph/vector store
+index = PropertyGraphIndex.from_existing(
+    property_graph_store=graph_store, vector_store=vector_store, ...
+)
+```
+
+### Construction
+
+Property graph construction in LlamaIndex works by performing a series of `kg_extractors` on each chunk, and attaching entities and relations as metadata to each llama-index node. You can use as many as you like here, and they will all get applied.
+
+If you've used transformations or metadata extractors with the [ingestion pipeline](../loading/ingestion_pipeline/index.md), then this will be very familiar (and these `kg_extractors` are compatible with the ingestion pipeline)!
+
+Extractors are set using the appropriate kwarg:
+
+```python
+index = PropertyGraphIndex.from_documents(
+    documents,
+    kg_extractors=[extractor1, extractor2, ...],
+)
+
+# insert additional documents / nodes
+index.insert(document)
+index.insert_nodes(nodes)
+```
+
+If not provided, the defaults are `SimpleLLMPathExtractor` and `ImplicitPathExtractor`.
+
+All `kg_extractors` are detailed below.
+
+#### (default) `SimpleLLMPathExtractor`
+
+Extract short statements using an LLM to prompt and parse single-hop paths in the format (`entity1`, `relation`, `entity2`)
+
+```python
+from llama_index.core.indices.property_graph import SimpleLLMPathExtractor
+
+kg_extractor = SimpleLLMPathExtractor(
+    llm=llm,
+    max_paths_per_chunk=10,
+    num_workers=4,
+    show_progress=False,
+)
+```
+
+If you want, you can also customize the prompt and the function used to parse the paths.
+
+Here's a simple (but naive) example:
+
+```python
+prompt = (
+    "Some text is provided below. Given the text, extract up to "
+    "{max_paths_per_chunk} "
+    "knowledge triples in the form of `subject,predicate,object` on each line. Avoid stopwords.\n"
+)
+
+
+def parse_fn(response_str: str) -> List[Tuple[str, str, str]]:
+    lines = response_str.split("\n")
+    triples = [line.split(",") for line in lines]
+    return triples
+
+
+kg_extractor = SimpleLLMPathExtractor(
+    llm=llm,
+    extract_prompt=prompt,
+    parse_fn=parse_fn,
+)
+```
+
+#### (default) `ImplicitPathExtractor`
+
+Extract paths using the `node.relationships` attribute on each llama-index node object.
```

Comment:
i'm actually not quite sure what's going on here. If node1 has node2 in its `relationships` with a `key`, then is the extracted tuple `(node1, relation, node2)`? 

(maybe i was confused by the "implicit" name, since the relationships are explicitly defined by the user :) )

Reply:
in any case worth adding a notebook here

Reply:
Haha I guess they are implicit in the sense that we aren't doing any calculation

The extraction is basically `(node1.node_id, relation, node2.node_id)` -- we store the text chunks in the graph as well, so we can link them